adapter/thing_management.md

```diff
@@ -0,0 +1,404 @@
+
+### Custom Inclusion
+
+Adapters utilizing custom protocol communication through local network or third-party cloud will have their own specific inclusion flow specified it in their own documentation.
+Preliminary steps of such flows will often include authentication with a third-party system, specific configuration, and selecting devices to be connected to the system.
+At the end of this process the adapter is always required to generate `evt.thing.inclusion_report` event with the thing specification.
+
+### Network Exclusion
+
+Network exclusion flow is used by wireless network adapters to exclude devices from the network and remove them from the system.
+
+* Client sends `cmd.thing.exclusion` command to the adapter to start exclusion process and switch adapter into unpairing mode.
+* User interacts with a physical device to switch it into unpairing mode, e.g. clicks a button on a device three times in a quick sequence.
+* At any time the adapter can generate `evt.thing.exclusion_status_report` event to report the current status of the exclusion process.
+* Once exclusion is finished adapter generates `evt.thing.exclusion_report` event with the excluded thing address.
+
+### Forced Exclusion
+
+If a device is offline and unreachable or the adapter does not support the network exclusion flow, the device can be force deleted from the system using this flow.
+
+* Client sends `cmd.thing.delete` command to the adapter to force delete a thing.
+* Adapter removes the thing from the network and generates `evt.thing.exclusion_report` event with the excluded thing address.
+
+### Requesting Inclusion Report
+
+* At any time a client can request inclusion report by sending `cmd.thing.get_inclusion_report` command to the adapter with a specified address of an already paired thing.
+* Adapter generates `evt.thing.inclusion_report` event with the thing specification.
+
+## FIMP Specification
+
+### Topics
+
+Every adapter should have unique resource name and use the following topics in accordance with [topic format](/fimp/topic_format.md):
+
+`pt:j1/mt:cmd/rt:ad/rn:{RESOURCE_NAME}/ad:1`
+
+`pt:j1/mt:evt/rt:ad/rn:{RESOURCE_NAME}/ad:1`
+
+### Service name
+
+Every adapter should define its own service name.
+
+### Interfaces
+
+| Type | Interface                         | Value type | Properties                                        | Description                                                                                                                 |
+|------|-----------------------------------|------------|---------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------|
+| in   | cmd.thing.inclusion               | bool       | `pin`, `force_non_secure`, `supports_pin_request` | For `true` starts and for `false` stops a thing inclusion process.                                                          |                         
```

Comment:
Question is... Do we want to notify mobile app about it that it sends also different list of props actually?
```
    "__interview__": "__interview__",
    "force_non_secure": "false",
    "pin": ""
```
Those are all props sent from mobile SDU while adding z-wave device.